include/camp/resource/stream.hpp

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright (c) 2016-18, Lawrence Livermore National Security, LLC.
+Produced at the Lawrence Livermore National Laboratory
+Maintained by Tom Scogland <scogland1@llnl.gov>
+CODE-756261, All rights reserved.
+This file is part of camp.
+For details about use and distribution, please read LICENSE and NOTICE from
+http://github.com/llnl/camp
+*/
+
+#ifndef __CAMP_STREAM_HPP
+#define __CAMP_STREAM_HPP
+
+
+#if defined(CAMP_HAVE_CUDA) || defined(CAMP_HAVE_HIP)
+namespace camp
+{
+namespace resources
+{
+  inline namespace v1
+  {
+
+    constexpr int DEFAULT_STREAM = -1;
```

Comment:
I like the constexpr ints here, but I'm thinking about the values a bit.  What do you think of making `DEFAULT_STREAM == 0`, then `NEXT_STREAM` keeps its default value, and we shift lookups by one, then provide this for all devices.  I ask this because it's conventional in a large number of places for 0 to be used as the default stream in cuda code, so it might help keep things consistent.

Reply:
I think I see what you mean, but just to confirm. If a user explicitly calls Cuda(1) they are actually given the 0th stream in the pool, Cuda(0) gives a default and Cuda(-1) would return the next stream in the pool?

Reply:
That's what I was thinking.  It's not objectively "better" but it is more like what cuda does by default, and I'm trying to figure out if that's what we want or not...